pkg/sources/reconciler/cloudeventssource/adapter.go

```diff
@@ -149,41 +164,27 @@ func makeAppEnv(o *v1alpha1.CloudEventsSource) []corev1.EnvVar {
 	return envs
 }
 
-// secretMountAtPath returns a build option for a service that adds a
-// secret based volume and mount a key at a path.
-func secretMountAtPath(name, mountPath, mountFile, secretName, secretKey string) resource.ObjectOption {
-	return func(object interface{}) {
-		ksvc, ok := object.(*servingv1.Service)
-		if !ok {
-			return
-		}
-
-		ksvc.Spec.Template.Spec.Volumes = append(
-			ksvc.Spec.Template.Spec.Volumes,
-			corev1.Volume{
-				Name: name,
-				VolumeSource: corev1.VolumeSource{
-					Secret: &corev1.SecretVolumeSource{
-						SecretName: secretName,
-						Items: []corev1.KeyToPath{{
-							Key:  secretKey,
-							Path: mountFile,
-						}},
-					},
-				},
-			})
-
-		if len(ksvc.Spec.Template.Spec.Containers) == 0 {
-			ksvc.Spec.Template.Spec.Containers = make([]corev1.Container, 1)
-		}
-
-		ksvc.Spec.Template.Spec.Containers[0].VolumeMounts = append(
-			ksvc.Spec.Template.Spec.Containers[0].VolumeMounts,
-			corev1.VolumeMount{
-				Name:      name,
-				ReadOnly:  true,
-				MountPath: mountPath,
+// secretVolumeAndMountAtPath returns a Secret-based volume and corresponding
+// mount at the given path.
+func secretVolumeAndMountAtPath(name, mountPath, mountFile, secretName, secretKey string) (corev1.Volume, corev1.VolumeMount) {
```

Comment:
This function was quite complex because it mixed structs instantiations with constructor logic that was conditionally creating containers, etc.
Now, it just returns the structs, and the object is constructed elsewhere.